### PR TITLE
Forget password by username from keycloak users api Task #202692

### PR DESCRIPTION
**Pull request by Tusharmahajan12**
# I have ensured that following `Pull Request Checklist` is taken care of before sending this PR

* [x] Code is formatted as per format decided
* [x] Updated acceptance criteria in tracker
* [x] Updated test cases in test-cases-tracker
* [x] Updated new API endpoint if any in common postman collection
* [x] Updated DB changes in db-tracker if any

### CODE REVIEW

File: src/src/services/keycloak/keycloak.service.ts [hunk 1]
```diff
@@ -116,14 +116,54 @@ export class KeycloakService {
 		}
 	}
 
+	public async getUserByUserName(username): Promise<{ [key: string]: any }> {
```
Comment (manojLondhe) [inline src/src/services/keycloak/keycloak.service.ts:119]: rename as

getUserByUsername

File: src/src/modules/auth/auth.service.ts [hunk 2]
```diff
@@ -103,17 +103,16 @@ export class AuthService {
 			});
 		}
 	}
-	public async resetPasswordUsingOtp(req, response) {
-		console.log('req', req);
+	public async getUserByusername(req) {
```
Comment (manojLondhe) [inline src/src/modules/auth/auth.service.ts:106]: add one blank line before start of new method

Comment (manojLondhe) [inline src/src/modules/auth/auth.service.ts:106]: rename as

getUserByUsername

**Review by manojLondhe — CHANGES_REQUESTED** [review]

File: src/src/services/keycloak/keycloak.service.ts (hunk 1 above)
Comment (Tusharmahajan12) [reply src/src/services/keycloak/keycloak.service.ts]: rename done

File: src/src/modules/auth/auth.service.ts (hunk 2 above)
Comment (Tusharmahajan12) [reply src/src/modules/auth/auth.service.ts:107]: done

Comment (Tusharmahajan12) [reply src/src/modules/auth/auth.service.ts:107]: rename done

File: src/src/modules/auth/auth.service.ts [hunk 3]
```diff
@@ -209,21 +219,7 @@ export class AuthService {
...
-                  id
-                  first_name
-                  mobile
-                }
-              }`,
-		};
-		const userRes: any = await this.hasuraService.postData(query);
-		console.log('userRes', userRes);
-
+		const userRes = await this.getUserByusername(req);
```
Comment (manojLondhe) [inline src/src/modules/auth/auth.service.ts:222]: not using renamed name?

File: src/src/modules/auth/auth.service.ts [hunk 4]
```diff
@@ -123,7 +123,17 @@ export class AuthService {
...
+		return userRes;
+	}
+
+	public async resetPasswordUsingOtp(req, response) {
+		console.log('req', req);
+		const username = req.username;
+		const hash = req.hash;
+		const otp = req.otp;
+		const reason = req.reason;
+		const userRes = await this.getUserByusername(req);
```
Comment (manojLondhe) [inline src/src/modules/auth/auth.service.ts:136]: not using renamed name?

File: src/src/modules/auth/auth.service.ts (hunk 3 above)
Comment (Tusharmahajan12) [reply src/src/modules/auth/auth.service.ts]: we have created new function getUserByusername and we are calling that so i had change there

File: src/src/modules/auth/auth.service.ts (hunk 4 above)
Comment (Tusharmahajan12) [reply src/src/modules/auth/auth.service.ts]: same here we created new function getUserByusername and using that

**Pull request merged**